### PR TITLE
ipwrap() Jinja filter

### LINKED ISSUE #61931: [FEATURE REQUEST] Jinja filter for wrapping IPv6 addresses in brackets ([ ])
**Is your feature request related to a problem? Please describe.**
A great deal of software that IPv6 addresses for configuration files, especially when a port or url is specified, to be wrapped in brackets ([ ]). There's currently no clean or easy way to do this (that I'm aware of )

**Describe the solution you'd like**
A jinja filter like Ansibles ipwrap() ( https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters_ipaddr.html#wrapping-ipv6-addresses-in-brackets )

**Describe alternatives you've considered**
Using something like the following:
```
{{ address if not address|is_ipv6 else ['[', address, ']']|join }}
```
But this will wrap IPv6 addresses with CIDR masks incorrectly. ie `2001:db8::1/64` becomes `[2001:db8::1/64]` instead of `[2001:db8::1]/64`.

### PR DESCRIPTION
**Pull request by ggiesen**
### What issues does this PR fix or reference?
Fixes: #61931

### New Behavior
Adds ipwrap() jinja filter to wrap IPv6 addresses with brackets.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by OrangeDog** [comment]
Would it make sense to implement this as an execution function rather than a filter? Then it can be used elsewhere too.

**Comment by MKLeb** [comment]
@ggiesen is this still WIP?

**Comment by MKLeb** [comment]
Since this PR was submitted, the `ip_bracket` util function was added, so I switched this over to using that.  @ggiesen any thoughts?  Still WIP to you?

File: doc/topics/jinja/index.rst [hunk 1]
```diff
@@ -1889,6 +1889,28 @@ Returns:
   ["fe80::"]
 
 
+.. jinja_ref:: ipwrap
+
+``ipwrap``
+----------
+
+.. versionadded:: 3005.0
```
Comment (Ch3LL) [inline doc/topics/jinja/index.rst:1897]:
```suggestion
.. versionadded:: 3006.0
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

**Review by MKLeb — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**